src/savefile/scene_serialize.c

```diff
@@ -447,9 +446,72 @@ void sceneAnimatorDeserialize(struct Serializer* serializer, struct Scene* scene
     }
 }
 
-#define INCLUDE_SAVEFILE_ALIGH_CHECKS   0
+void securityCameraSerialize(struct Serializer* serializer, SerializeAction action, struct Scene* scene) {
+    u8 serializedCount = 0;
+    short heldCam = -1;
+    for (int i = 0; i < scene->securityCameraCount; ++i) {
+        struct SecurityCamera* cam = &scene->securityCameras[i];
+        if (!(cam->rigidBody.flags & RigidBodyIsKinematic)) {
+            if (&cam->collisionObject == scene->player.grabConstraint.object) {
+                heldCam = serializedCount;
+            }
+            ++serializedCount;
+        }
+    }
+    action(serializer, &serializedCount, sizeof(u8));
+    action(serializer, &heldCam, sizeof(short));
+    
+    for (int i = 0; i < scene->securityCameraCount; ++i) {
+        struct SecurityCamera* cam = &scene->securityCameras[i];
+        if (!(cam->rigidBody.flags & RigidBodyIsKinematic)) {
+            u8 index = i;
+            action(serializer, &index, sizeof(u8));
+            
+            action(serializer, &cam->rigidBody.transform, sizeof(struct PartialTransform));
+            action(serializer, &cam->rigidBody.velocity, sizeof(struct Vector3));
+            action(serializer, &cam->rigidBody.angularVelocity, sizeof(struct Vector3));
+            action(serializer, &cam->rigidBody.flags, sizeof(enum RigidBodyFlags));
+            action(serializer, &cam->rigidBody.currentRoom, sizeof(short));
+        }
+    }
+}
+
+void securityCameraDeserialize(struct Serializer* serializer, struct Scene* scene) {
+    u8 serializedCount;
+    serializeRead(serializer, &serializedCount, sizeof(u8));
+    
+    short heldCam;
+    serializeRead(serializer, &heldCam, sizeof(short));
+    
+    for (int i = 0; i < serializedCount; ++i) {
+        u8 index;
+        serializeRead(serializer, &index, sizeof(u8));
+        if (index >= scene->securityCameraCount) {
+            continue;
+        }
+        
+        struct SecurityCamera* cam = &scene->securityCameras[index];
+        
+        rigidBodyUnmarkKinematic(&cam->rigidBody, SECURITY_CAMERA_RIGID_BODY_MASS, securityCameraMofI());
```

Comment:
Nice.

Rather than exposing security camera implementation details here, can you create some helper functions in `security_camera` like:
```c
int securityCameraIsDetached();
void securityCameraDetach();
```

That way you can keep things more self-contained, avoid the need for the public constant and new helper function, and better ensure consistency between serialization/deserialization and runtime behavior.

Similar to the pattern followed by ball catchers, which have `ballCatcherHandBall()` for this reason.

Looks good otherwise.

Reply:
Yeah, it's definitely cleaner this way.